src/components/Button/Button.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import ButtonType from './ButtonType';
+import './Button.scss';
+
+const propTypes = {
+    /** Content of the button, e.g. a string label or more complex React components */
+    children: PropTypes.node.isRequired,
+    /** Extra classes for the top-level wrapper */
+    className: PropTypes.string,
+    /** for styling purposes */
+    isActive: PropTypes.bool,
+    /** set disabled HTML attribute, custom styling and will prevent onClick from firing */
+    isDisabled: PropTypes.bool,
+    isFullWidth: PropTypes.bool,
+    onClick: PropTypes.func,
+    /** object with camelCased CSS rules, e.g.  style={{ marginTop: -1}} */
+    style: PropTypes.objectOf(PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.number,
+    ])),
+    type: PropTypes.oneOf([
+        ButtonType.CLEAR,
+        ButtonType.DEFAULT,
+        ButtonType.PRIMARY,
+    ]),
+};
+
+const defaultProps = {
+    className: '',
+    isActive: false,
+    isDisabled: false,
+    isFullWidth: false,
+    onClick: null,
+    style: {},
+    type: ButtonType.DEFAULT,
+};
+
+class Button extends Component {
```

Comment:
Are we going to support the confirm/reject behaviour in the existing button?

Reply:
Added `hasConfirm` to support the confirmation state